packages/codemods/src/v4-to-v5/transformers/machine-to-create-machine.ts

```diff
@@ -0,0 +1,48 @@
+import ts from 'typescript';
+import {
+  isMachineCallExpression,
+  isMachineNamedImportSpecifier,
+  isMachinePropertyAccessExpression,
+} from '../predicates';
+
+export const machineToCreateMachine: ts.TransformerFactory<ts.SourceFile> =
+  (context) => (sourceFile) => {
+    const { factory } = context;
+    return ts.visitNode(sourceFile, visit);
+
+    function visit(node: ts.Node): ts.Node {
+      if (isMachineNamedImportSpecifier(node)) {
+        // if we have `propertyName`, it's using `as` syntax so we need to
+        // rename `propertyName`. otherwise we just rename `name`.
+        const [propertyName, name] = node.propertyName
+          ? [factory.createIdentifier('createMachine'), node.name]
+          : [node.propertyName, factory.createIdentifier('createMachine')];
+
+        return factory.updateImportSpecifier(
+          node,
+          node.isTypeOnly,
+          propertyName,
+          name,
+        );
+      }
+
+      if (isMachineCallExpression(node)) {
+        return factory.updateCallExpression(
+          node,
+          factory.createIdentifier('createMachine'),
+          node.typeArguments,
+          node.arguments,
+        );
+      }
+
+      if (isMachinePropertyAccessExpression(node)) {
+        return factory.updatePropertyAccessExpression(
+          node,
+          node.expression,
+          factory.createIdentifier('createMachine'),
+        );
+      }
+
+      return ts.visitEachChild(node, visit, context);
```

Comment:
`visitEachChild` calls `visit` on each of the node's child nodes. This allows us to make `visit` _recursive_.

For each node, we either return an updated version of it (if one of the `if` blocks apply for that node) or we visit each of the node's children and return them. This allows us to traverse through (nearly) every node in the tree, just by returning `ts.visitEachChild`!